Graphd crashed executing match query

**Your Environments (__required__)**
nebula 3.5 d1e2118

**How To Reproduce(__required__)**
use nba, graph crashes when the following two statements are executed(sometimes it may be necessary to repeat these two statements to reproduce) :

```
match (v.player) return v
match (v:player) where v.player.name == "Tim Duncan" return v;
```

```

(root@nebula) [(none)]> use nba
Execution succeeded (time spent 782µs/14.696875ms)

Mon, 22 May 2023 10:01:58 CST

(root@nebula) [nba]> show tag indexes
+---------------------+----------+----------+
| Index Name          | By Tag   | Columns  |
+---------------------+----------+----------+
| "player_age_index"  | "player" | ["age"]  |
| "player_name_index" | "player" | ["name"] |
| "team_name_index"   | "team"   | ["name"] |
+---------------------+----------+----------+
Got 3 rows (time spent 597µs/15.761625ms)

Mon, 22 May 2023 10:02:03 CST

(root@nebula) [nba]> match (v.player) return v
[ERROR (-1004)]: SyntaxError: Invalid node pattern near `(v.player)'

Mon, 22 May 2023 10:02:06 CST

(root@nebula) [nba]> match (v:player) where v.player.name == "Tim Duncan" return v;
[ERROR (-1005)]: Error found in optimization stage: The output columns of the OptGroupNode is different from OptGroup when applying the rule: RemoveNoopProjectRule, (null) vs. __IndexScan_1

Mon, 22 May 2023 10:02:08 CST

(root@nebula) [nba]> match (v:player) where v.player.name == "Tim Duncan" return v;
2023/05/22 10:02:09 Loop error, failed to reconnect, failed to open transport, error: dial tcp 192.168.15.31:10050: connect: connection refused

Bye root!
Mon, 22 May 2023 10:02:09 CST

2023/05/22 10:02:09 [WARNING] Sign out failed, write tcp 10.8.0.26:60744->192.168.15.31:10050: write: broken pipe
panic: Loop error, failed to reconnect, failed to open transport, error: dial tcp 192.168.15.31:10050: connect: connection refused

goroutine 1 [running]:
log.Panicf(0x13815e8, 0xe, 0xc0000a9e38, 0x1, 0x1)
	/opt/hostedtoolcache/go/1.16.4/x64/src/log/log.go:361 +0xc5
main.main()
	/home/runner/work/nebula-console/nebula-console/main.go:569 +0x5c8
```

core dump
```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
--Type <RET> for more, q to quit, c to continue without paging--
Core was generated by `bin/nebula-graphd -flagfile /data2/luban/knife/nebula-ent-3.5.0-rc/etc/graphd2/'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000001259776 in std::_Function_handler<void (void*), nebula::ObjectPool::OwnershipHolder::OwnershipHolder<nebula::ConstantExpression>(nebula::ConstantExpression*)::{lambda(void*)#1}>::_M_invoke(std::_Any_data const&, void*&&) ()
[Current thread is 1 (Thread 0x7f58073ff700 (LWP 3354998))]
(gdb) bt
#0  0x0000000001259776 in std::_Function_handler<void (void*), nebula::ObjectPool::OwnershipHolder::OwnershipHolder<nebula::ConstantExpression>(nebula::ConstantExpression*)::{lambda(void*)#1}>::_M_invoke(std::_Any_data const&, void*&&) ()
#1  0x000000000126501d in std::__cxx11::_List_base<nebula::ObjectPool::OwnershipHolder, std::allocator<nebula::ObjectPool::OwnershipHolder> >::_M_clear() ()
#2  0x00000000015f0947 in nebula::graph::QueryContext::~QueryContext() ()
#3  0x00000000015f0b39 in nebula::graph::QueryContext::~QueryContext() ()
#4  0x00000000012b6424 in nebula::graph::QueryInstance::onError(nebula::Status) ()
#5  0x00000000012b8da4 in nebula::graph::QueryInstance::execute() ()
#6  0x00000000012b2f05 in nebula::graph::QueryEngine::execute(std::unique_ptr<nebula::graph::RequestContext<nebula::ExecutionResponse>, std::default_delete<nebula::graph::RequestContext<nebula::ExecutionResponse> > >) ()
#7  0x00000000012832b6 in ?? ()
#8  0x00000000012839dc in ?? ()
#9  0x000000000254586c in ?? ()
#10 0x00000000022d72a7 in virtual thunk to apache::thrift::concurrency::FunctionRunner::run() ()
#11 0x0000000002416678 in apache::thrift::concurrency::ThreadManager::Impl::Worker::run() ()
#12 0x000000000241877e in apache::thrift::concurrency::PthreadThread::threadMain(void*) ()
#13 0x00007f5841183609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#14 0x00007f58410a8133 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Expected behavior**
graphd won't crash.

--- a/src/parser/parser.yy
+++ b/src/parser/parser.yy
@@ -1801,12 +1801,10 @@ match_node
         $$ = new MatchNode();
     }
     | parenthesized_expression {
-        auto& e = $1;
-        if (e->kind() != Expression::Kind::kLabel) {
-            delete $1;
+        if ($1->kind() != Expression::Kind::kLabel) {
             throw nebula::GraphParser::syntax_error(@1, "Invalid node pattern");
         }
-        $$ = new MatchNode(static_cast<LabelExpression*>(e)->name(), nullptr, nullptr);
+        $$ = new MatchNode(static_cast<LabelExpression*>($1)->name(), nullptr, nullptr);
     }
     | L_PAREN match_alias match_node_label_list R_PAREN {
         $$ = new MatchNode(*$2, $3, nullptr);

--- a/tests/tck/features/bugfix/SyntaxErrorCrash.feature
+++ b/tests/tck/features/bugfix/SyntaxErrorCrash.feature
@@ -1,7 +1,7 @@
 # Copyright (c) 2023 vesoft inc. All rights reserved.
 #
 # This source code is licensed under Apache 2.0 License.
-# #2780 ent
+# #5558
 Feature: Test crash double delete expr
 
   Background:

--- a/tests/tck/features/bugfix/SyntaxErrorCrash.feature
+++ b/tests/tck/features/bugfix/SyntaxErrorCrash.feature
@@ -1,0 +1,32 @@
+# Copyright (c) 2023 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License.
+# #2780 ent
+Feature: Test crash double delete expr
+
+  Background:
+    Given a graph with space named "nba"
+
+  Scenario: crash double delete expr
+    When executing query:
+      """
+      match (v.player) return v;
+      """
+    Then a SyntaxError should be raised at runtime:  Invalid node pattern near `(v.player)'
+    When executing query:
+      """
+      match (v.player) return v;
+      """
+    Then a SyntaxError should be raised at runtime:  Invalid node pattern near `(v.player)'
+    When executing query:
+      """
+      match (v.player) return v;
+      """
+    Then a SyntaxError should be raised at runtime:  Invalid node pattern near `(v.player)'
+    When executing query:
+      """
+      return 1;
+      """
+    Then the result should be, in any order:
+      | 1 |
+      | 1 |

